src/unix/process.c

```diff
@@ -505,6 +505,9 @@ int uv_spawn(uv_loop_t* loop,
   if (exec_errorno == 0) {
     QUEUE_INSERT_TAIL(&loop->process_handles, &process->queue);
     uv__handle_start(process);
+  } else {
+    err = exec_errorno;
+    goto error;
```

Comment:
We cannot jump to `error` here, because the stdio streams need to be closed properly. Check the for loop above.